src/error/mod.rs

```diff
@@ -297,6 +297,22 @@ impl Error {
             ])
     }
 
+    pub(crate) fn argument_subcommand_conflict(
```

Comment:
I'm hijacking the `ArgumentConflict` error here, it would probably make sense to introduce a new error kind specifically for this, but I didn't want to spend hours on error boilerplate with a PR without first getting some feedback on the general approach.